docs/releases.md

```diff
@@ -0,0 +1,71 @@
+# Ceph CSI driver Release Process
+
+- [Ceph CSI driver Release Process](#Ceph-csi-driver-release-process)
+  - [Introduction](#introduction)
+  - [Versioning](#versioning)
+  - [Tagging repositories](#tagging-repositories)
+  - [Release process](#release-process)
+
+## Introduction
+
+This document provides details about Ceph CSI driver releases.
+
+## Versioning
+
+The Ceph CSI driver project uses
+[semantic versioning](http://semver.org/)
+for all releases.
+Semantic versions are comprised of three
+fields in the form:
+
+```MAJOR.MINOR.PATCH```
+
+For examples: `1.0.0`, `1.0.0-rc.2`.
+
+Semantic versioning is used since the version
+number is able to convey clear information about
+how a new version relates to the previous version.
+For example, semantic versioning can also provide
+assurances to allow users to know when they must
+upgrade compared with when they might want to upgrade:
+
+- When `PATCH` increases, the new release contains important **security fixes**,
+general bug fixes  and an upgrade is recommended.
+
+ The patch field can contain extra details after the number.
+ Dashes denote pre-release versions.`1.0.0-rc.2` in the example
+ denotes the second release candidate for release `1.0.0`.
+
+- When `MINOR` increases, the new release adds **new features**
+and it must be backward compatible.
+
+- When `MAJOR` increases, the new release adds **new features,
+  bug fixes, or both** and which *changes the behavior from
+  the previous release* (may be backward incompatible).
+
+## Tagging repositories
+
+The tag name must begin with "v" followed by the version number, conforming to
+the [versioning](#versioning) requirements (e.g. a tag of `v1.0.0-rc2` for
+version `1.0.0-rc2`). This tag format is used by the Travis CI infrastructure to
+properly upload and tag releases to Quay and Docker Hub.
+
+## Release process
```

Comment:
Comments for this section:
- Release checklist is a list of issues/PRs that are to be completed before the release is made?
- How can a user relate to a release and figure out what its contents are? IOW, release notes or how to determine an issue or a PR is part of a release?

Reply:
I have added some more details to address above queries. PTAL.

Reply:
This is still inadequate, or I am unable to understand how things will move.

As this PR intends to start a release process doc. and update its contents as we progress towards a release, I would suggest adding a TBD here around the various bullets and as time progresses updating the same.

Reply:
Added `TBD` to the section, so it convey it can be imrpoved any time.